### PR TITLE
[SandboxIR] Implemented isVolatile() for LoadInst

### PR DESCRIPTION
**Pull request by wizardengineer**
as titled stated

cc: @vporpo

### CODE REVIEW

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp [hunk 1]
```diff
@@ -737,7 +737,7 @@ define void @foo(i1 %cond0, i1 %cond2) {
 TEST_F(SandboxIRTest, LoadInst) {
   parseIR(C, R"IR(
 define void @foo(ptr %arg0, ptr %arg1) {
-  %ld = load i8, ptr %arg0, align 64
+  %ld = load volatile i8, ptr %arg0, align 64
```
Comment (vporpo) [inline llvm/unittests/SandboxIR/SandboxIRTest.cpp:740]: Nit: Let's keep both the original non-volatile load and the volatile load, and `EXPECT_FALSE(Ld->isVolatile())` for the first one and `EXPECT_TRUE(VolLd->isVolatile())`.

**Comment by vporpo** [comment]
Now that we can check `isVolatile()` we can also add a new `LoadInst::create()` function with a `bool isVolatile` argument. But this can be a separate PR.

File: llvm/unittests/SandboxIR/SandboxIRTest.cpp (hunk 1 above)
Comment (medievalghoul) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp:740]: Ah okay, that makes sense. So the end result would be like this?

```
%ld = load i8, ptr %arg0, align 64
%vld = load volatile i8, ptr %arg0, align 64
```

Comment (vporpo) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp:740]: yes, this looks great.

Comment (medievalghoul) [reply llvm/unittests/SandboxIR/SandboxIRTest.cpp]: I fixed it

**Comment by medievalghoul** [comment]
@vporpo added the fix :)

**Review by vporpo — APPROVED** [review]
Looks good, thanks!

**Comment by medievalghoul** [comment]
> Now that we can check `isVolatile()` we can also add a new `LoadInst::create()` function with a `bool isVolatile` argument. But this can be a separate PR.

@vporpo working on it now, should i include the body of the function too or will the function header alone suffice for now?

**Comment by vporpo** [comment]
> should i include the body of the function too or will the function header alone suffice for now?
I think you should include the body too + test it. 
The best way to implement this is probably to add the `isVolatile` argument to the existing create() functions, then add new create() functions that look just like the existing ones (without the `isVolatile` argument) and in their bodies call the ones with the argument but set it to false.

**Comment by medievalghoul** [comment]
@aeubanks @tmsri is this good to merge?

**Comment by vporpo** [comment]
> @aeubanks @tmsri is this good to merge?

Go ahead, it looks good.

**Review by tmsri — APPROVED** [review]
LGTM

**Comment by medievalghoul** [comment]
> Go ahead, it looks good.
@vporpo I don't have write access, you guys will have to :)

**Comment by vporpo** [comment]
Ah OK, let me merge it then.

**Comment by vporpo** [comment]
There is some trailing whitespace in two lines. I will push a fix, but why didn't the format checker bot catch it? Or did I miss it?

**Pull request merged**